api/src/main/resources/application_detailed_summary.py

```diff
@@ -0,0 +1,101 @@
+import time
+import json
+import logging
+import sys
+from importlib import import_module
+
+from summary_aggregator import ComponentSummaryAggregator
+from plugins_summary.yarn_connection import YarnConnection
+from async_dispatcher import AsyncDispatcher
+import application_registrar
+import application_summary_registrar
+import deployer_utils
+
+class ApplicationDetailedSummary(object):
+
+    def __init__(self, environment, config):
+        self._environment = environment
+        self._configs = config
+        self._application_registrar = \
+        application_registrar.HbaseApplicationRegistrar(environment['hbase_thrift_server'])
+        self._application_summary_registrar = \
+        application_summary_registrar.HBaseAppplicationSummary(environment['hbase_thrift_server'])
+        self._yarn_connection = \
+        YarnConnection(environment['yarn_resource_manager_host'], \
+            environment['yarn_resource_manager_port'])
+        self._summary_aggregator = ComponentSummaryAggregator()
+        self._component_creators = {}
+        self.dispatcher = AsyncDispatcher()
```

Comment:
In the last version of this you were using a pool of 4 workers. This creates a pool of 50 workers.

Reply:
Earlier with 4 processes, we had 60 seconds of sleep after one round of generating the summary for all apps. This had to be optimized.
Now, assuming 100 apps with each app taking about 6 secs to process the summary, based on our analysis, it was appropriate to create around 50 threads with a sleep of about 15 seconds after one round of generating summary for all apps.

Reply:
No, you can't have 50 threads to do this. Give it a maximum of 5 threads and arrange things so the update frequency *fairly* decreases for *all* applications proportionally.